src/algorand_transactions/transactions_group.rs

```diff
@@ -0,0 +1,240 @@
+use serde::{Deserialize, Serialize};
+use sha2::Digest;
+
+use crate::{
+    algorand_errors::AlgorandError,
+    algorand_types::{Byte, Bytes, Result},
+    AlgorandHash,
+    AlgorandKeys,
+    AlgorandSignedTransaction,
+    AlgorandTransaction,
+};
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct TxGroup {
+    #[serde(rename = "txlist", default)]
+    tx_group_hashes: Vec<AlgorandHash>,
+
+    #[serde(skip_serializing)]
+    txs: Vec<AlgorandTransaction>,
+
+    #[serde(skip_serializing)]
+    group_id: Option<AlgorandHash>,
+}
+
+impl TxGroup {
+    const MAX_TX_GROUP_SIZE: usize = 16;
+
+    pub fn new(transactions: &[AlgorandTransaction]) -> Result<TxGroup> {
+        let a = transactions.iter().map(|tx| tx.to_raw_tx_id()).collect();
+        match a {
+            Ok(res) => Ok(Self {
+                txs: transactions.to_vec(),
+                group_id: None,
+                tx_group_hashes: res,
+            }),
+            _ => Err(AlgorandError::Custom("error".to_string())),
+        }
+    }
+
+    pub fn assign_group_id(&mut self) -> Result<AlgorandHash> {
+        let group_id = TxGroup::compute_group_id(self)?;
+        self.txs = self
+            .txs
+            .iter()
+            .map(|tx| tx.assign_group_id(group_id))
+            .collect();
+        self.group_id = Some(group_id);
+        Ok(group_id)
+    }
+
+    pub fn compute_group_id(&self) -> Result<AlgorandHash> {
+        if self.tx_group_hashes.is_empty() {
+            return Err(AlgorandError::Custom("Empty transactions list".to_string()));
```

Comment:
Another usage of `AlgorandError::Custom`...

Reply:
See my review (once done).